executor/src/witgen/generator.rs

```diff
@@ -181,7 +195,9 @@ impl<'a, T: FieldElement> Generator<'a, T> {
         );
         processor.solve(&mut sequence_iterator).unwrap();
 
-        processor.finish().remove(1)
+        // Ignore any updates to the publics at this point, as we'll re-visit the last row again.
```

Comment:
No. This piece of code is just to initialize the first row from identities like `pc' = (1 - first_step') * <...>`.

Since this is the very first thing that happens, we won't have any publics at that point. We could in theory *derive* a public value here (e.g. if there was a constraint like `first_step' * (foo - :final_foo) = 0`, but we'll process the last row again anyway, so it should be OK to discard here.